packages/fx-core/src/core/middleware/projectMigrator.ts

```diff
@@ -368,6 +401,10 @@ async function generateArmParameterJson(ctx: CoreHookContext) {
   const fx = path.join(inputs.projectPath as string, `.${ConfigFolderName}`);
   const fxConfig = path.join(fx, InputConfigsFolderName);
   const envConfig = await fs.readJson(path.join(fx, "env.default.json"));
+  const hasProvision = envConfig.solution?.provisionSucceeded as boolean;
```

Comment:
the provision flag may be false when partially provision the resource. For example, user provisioned tab app first, then adds a bot to the project. Our tooling will set provision success to false again in this case. Can you do not check this flag? Just see whether there're resource names in env.default.json, and fill the existing names to parameters.dev.json.

Reply:
sure

Reply:
add on message

Reply:
yes, the message changed